### DISCUSSION TITLE
New views layout for greater focus and productivity

### DISCUSSION THREAD

**Discussion by d13** [gitkraken/vscode-gitlens#2721] [Announcements]
In GitLens 14, we've restructured our views for greater focus and productivity, including the new _GitLens Inspect_ side bar. We also moved some of our views from the Source Control side bar into either the GitLens or the new GitLens Inspect side bar.

![Screenshot with new layout](https://github.com/gitkraken/vscode-gitlens/assets/293282/09f57434-3946-42c1-b3e6-36d0844b136b)

> GitLens Inspect side bar as shown above has been manually moved into the Secondary Side Bar (see pro-tip below)

## GitLens Inspect side bar

The all new GitLens Inspect is like an x-ray into your code, which focuses on providing contextual information and insights to what you're actively working on. Think about it as the developer tools inspector for your code.

Views include:

- Commit Details
- Line History
- File History
- Visual File History
- Search & Compare

> Pro tip: you can drag the entire GitLens Inspect side bar onto your secondary side bar, without needing to arrange views on-by-one

## GitLens side bar

We've re-focused this to be the home of GitKraken teams and collaboration services (e.g. GitKraken Workspaces) as well as discovery, help, and support.

Views include:

- Home view
- GitKraken Workspaces
- GitKraken Account

## Source Control side bar

We've slimmed down this side bar to only contain views that are focused on your repositories.

Views include:

- Commits 
- Branches
- Remotes
- Stashes
- Tags
- Worktrees
- Contributors
- Repositories _(hidden by default)_

## (Bottom) Panel

The new home for the Commit Graph which provides convenient and easy access to the Graph. Additionally, this includes a dedicated details view. Quickly toggle the Graph via the new `Toggle Commit Graph` command.

Views include:

- Commit Graph
- Commit Graph Details

> Pro tip: Setup a hotkey to toggle the Graph by opening the Command Palette, type `Toggle Commit Graph`, and click the "cog" icon on the end. When the Keyboard Shortcuts tab opens, press the "add" icon and record a keybinding. 

## Updating to the new views layout

Unless you opt into the new layout, which we recommend, your existing views will stay where they are. You can opt-in at any time from `Reset Views Layout` in the Command Palette. 

After upgrading to GitLens 14 (or June 8th's Pre-Release), you'll be prompted to change to this layout:

![Screenshot of change layout prompt](https://github.com/gitkraken/vscode-gitlens/assets/293282/c5342d3f-6a41-4367-b914-f1c37102d4aa)

**Comment by saatsazov** [comment]
I would like to know how to disable this annoying popup. I open every project and it asks me every time to try new layout. Thank you.

**Reply by jkelroy** [reply]
Hello @saatsazov, this was not intentional and has been addressed  (#2725) in the latest pre-release. 

You can switch to the latest pre-release by going to Extensions and selecting `Update` on GitLens. 

<img width="450" alt="Screenshot 2023-06-08 at 9 53 40 AM" src="https://github.com/gitkraken/vscode-gitlens/assets/27739784/8557ae2f-23e6-4751-abc2-8e4f2ef6e17f">

**Comment by devinrhode2** [comment]
A video demo of this would be awesome! I have my set patterns of usage for gitlens.

Basically, I have my traditional vscode source control/git pane.

In that same side bar on the bottom half, I have gitlens "Commits" view.

As I commit, changes move downward.

I often use `fixup! Foo`, `amend! Foo .. Bar` commits, with `rebase.autoSquash` enabled, and am running `git pull origin master:master --rebase=i` to pull latest master. (`:master` causes local master to update, `=i` means interactive, which causes auto-squashing to happen).

`git absorb` would be really good to add into git lens ui. I've outlined that in another issue!

**Reply by devinrhode2** [reply]
I often use gitlen's git blame to copy-paste a commit message into `fixup! Foo` commits that I manually write.

Would love some options to "squash/fixup/amend" in gitlens' blame hovercard.

Any action that creates a squash/fixup/amend commit could immediately launch into rebasing that change.

Rebasing a singular change is easier than rebasing multiple different fixup/squashes. (e.g. You know you just renamed A to B or changed interface Y to Z, now handle conflicts for that change and that change only)

**Reply by devinrhode2** [reply]
PR autolinks are super nice - if any SCM providers are not supported I'm sure people would love to see that added.

**Reply by devinrhode2** [reply]
My "Commits" pane disappeared, but I was able to go into here, Click "Commits" and restored the pane by dragging it back to source control pane and dropping it on the bottom half.
![ScreenShot 2023-06-14 at 22 27 03@2x](https://github.com/gitkraken/vscode-gitlens/assets/539816/59cdac9b-5b77-431d-a5e2-4787e6489e2f)

**Reply by eamodio** [reply]
> My "Commits" pane disappeared, but I was able to go into here, Click "Commits" and restored the pane by dragging it back to source control pane and dropping it on the bottom half. ![ScreenShot 2023-06-14 at 22 27 03@2x](https://user-images.githubusercontent.com/539816/245984824-59cdac9b-5b77-431d-a5e2-4787e6489e2f.png)

I don't understand how Commits would have disappeared (or moved) -- its default "home" wasn't changed. FWIW, you can also get access to any of the views (hidden or not) from the Command Palette.

**Comment by adalisk-emikhaylov** [comment]
> Unless you opt into the new layout, which we recommend, your existing views will stay where they are. 

I don't think I got a prompt. My views moved themselves from the "source control" tab to the new tabs.

**Comment by rocka** [comment]
Can the views under "Source Control side bar" be moved to "GitLens side bar" or "GitLens Inspect side bar", like "GitLens Layout" in version 13.x? I have many submodules in a project, and the Source Control side bar is too crowded.

**Reply by eamodio** [reply]
Yes, though you'll need to drag and drop them there manually.

**Comment by jacob121532** [comment]
Bs
